src/vec/FixedPairList.cpp

```diff
@@ -0,0 +1,345 @@
+    sigBeforeSend.disconnect();
+    sigAfterRecv.disconnect();
+    sigOnParticlesChanged.disconnect();
+  }
+
+  real FixedPairList::getLongtimeMaxBondSqr()
+  {
+    return longtimeMaxBondSqr;
+  }
+
+  void FixedPairList::setLongtimeMaxBondSqr(real d)
+  {
+    longtimeMaxBondSqr = d;
+  }
+
+  void FixedPairList::resetLongtimeMaxBondSqr()
+  {
+    longtimeMaxBondSqr = 0.0;
+  }
+
+  bool FixedPairList::add(size_t pid1, size_t pid2)
+  {
+    bool returnVal = true;
+    if (pid1 > pid2)
+      std::swap(pid1, pid2);
+
+    auto const& storageVec = vectorization->storageVec;
+    size_t const p1 = storageVec->lookupRealParticleVec(pid1);
+    size_t const p2 = storageVec->lookupLocalParticleVec(pid2);
+
+    if(p1==VEC_PARTICLE_NOT_FOUND) {
+      // Particle does not exist here, return false
+      returnVal=false;
+    } else {
+      if(p2==VEC_PARTICLE_NOT_FOUND) {
+        LOG4ESPP_DEBUG(theLogger, "Particle p2 " << pid2 << " not found");
+      }
+    }
+
+    if(returnVal){
+      // add the pair locally
+      this->push_back({p1,p2});
+
+      // ADD THE GLOBAL PAIR
+      // see whether the particle already has pairs
+      std::pair<GlobalPairs::const_iterator,
+        GlobalPairs::const_iterator> equalRange
+        = globalPairs.equal_range(pid1);
+      if (equalRange.first == globalPairs.end()) {
+        // if it hasn't, insert the new pair
+        globalPairs.insert(std::make_pair(pid1, pid2));
+      }
+      else {
+        // otherwise test whether the pair already exists
+        for (GlobalPairs::const_iterator it = equalRange.first; it != equalRange.second; ++it) {
+          if (it->second == pid2) {
+            // Pair already exists, generate error!
+            throw std::runtime_error("Duplicate pair inserted");
+            ;
+          }
+        }
+        // if not, insert the new pair
+        globalPairs.insert(equalRange.first, std::make_pair(pid1, pid2));
+      }
+      LOG4ESPP_INFO(theLogger, "added fixed pair to global pair list");
+    }
+    LOG4ESPP_DEBUG(theLogger, "Leaving add with returnVal " << returnVal);
+    return returnVal;
+  }
+
+  python::list FixedPairList::getBonds()
+  {
+    python::tuple bond;
+    python::list bonds;
+    for (GlobalPairs::const_iterator it=globalPairs.begin(); it != globalPairs.end(); it++) {
+      bond = python::make_tuple(it->first, it->second);
+      bonds.append(bond);
+    }
+
+    return bonds;
+  }
+
+  std::vector<size_t> FixedPairList::getPairList()
+  {
+    std::vector<size_t> ret;
+    for (GlobalPairs::const_iterator it = globalPairs.begin(); it != globalPairs.end(); it++) {
+      ret.push_back(it->first);
+      ret.push_back(it->second);
+    }
+    return ret;
+  }
+
+  python::list FixedPairList::getAllBonds()
+  {
+    std::vector<size_t> local_bonds;
+    std::vector<std::vector<size_t> > global_bonds;
+    python::list bonds;
+
+    for (GlobalPairs::const_iterator it = globalPairs.begin(); it != globalPairs.end(); it++) {
```

Comment:
I prefer `auto` for iterators. This also applies for the other iterator-based loops.
```suggestion
    for (auto it = globalPairs.cbegin(); it != globalPairs.cend(); it++) {
```